[Excel] (Custom functions) Update low engagement articles (#5364)
OfficeDev/office-js-docs-pr: docs/excel/custom-functions-batching.md

# Batch custom function calls for a remote service

> Use batching to group calls to a remote service into one network request. This cuts down the number of network round trips to your remote service and helps the worksheet finish recalculating faster.

Reviewer: ```suggestion
Use batching to group calls to a remote service into a single network request. This reduces the number of network round trips to your remote service and helps your worksheet finish recalculating faster.
```

> Here's a batching example scenario: 100 cells call the custom function. Instead of 100 network requests, you send one request that lists all 100 operations and then returns 100 answers.

Reviewer: ```suggestion
Here's a batching example scenario: If 100 cells call the custom function, send one request that lists all 100 operations. The service returns 100 results in a single response.
```

## Key points

To set up batching for your custom functions you'll need to write three main sections of code.

1. A [push operation](#add-the-_pushoperation-function) to add a new operation to the batch of calls each time Excel calls your custom function.
2. A [function to make the remote request](#make-the-remote-request) when the batch is ready.
3. [Server code to respond to the batch request](#process-the-batch-call-on-the-remote-service), calculate all of the operation results, and return the values.

[!include[Excel custom functions note](../includes/excel-custom-functions-note.md)]

## Create the batching pattern in this article

In the following sections, you'll learn how to construct the code one example at a time. It's recommended you create a brand-new custom functions project using the [Yeoman generator for Office Add-ins](../develop/yeoman-generator-overview.md) generator. To create a new project, see [Get started developing Excel custom functions](../quickstarts/excel-custom-functions-quickstart.md). You can use TypeScript or JavaScript.

> [!TIP]

> > To view the completed sample, create a new custom functions project with the Yeoman generator for Office Add-ins, and then paste the code examples into your own project. Run the code and try it out.

Reviewer: ```suggestion
> To view the completed sample, create a new custom functions project with the Yeoman generator for Office Add-ins. Copy and paste the code examples into your project then, run the code and try it out.
```

>
> Alternatively, download or view the complete sample project at [Custom function batching pattern](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Excel-custom-functions/Batching). If you want to view the code in whole before reading any further, take a look at the [script file](https://github.com/OfficeDev/Office-Add-in-samples/blob/main/Excel-custom-functions/Batching/src/functions/functions.js).

## Batch each call to your custom function

> Your custom functions work by calling a remote service to perform the operation and calculate the result they need. This provides a way for them to store each requested operation into a batch. Later you'll see how to create a `_pushOperation` function to batch the operations. First, take a look at the following code example to see how to call `_pushOperation` from your custom function.

Reviewer: ```suggestion
Your custom functions work by calling a remote service to perform operations and return results. This provides a way for them to store each requested operation into a batch. Later you'll see how to create a `_pushOperation` function to batch the operations. First, review the following code example to see how to call `_pushOperation` from your custom function.
```

# Batch custom function calls for a remote service

Use batching to group calls to a remote service into one network request. This cuts down the number of trips to your remote service and helps the worksheet finish recalculating faster.

> ## Key points

Reviewer 2: After reading through all these bullet points, it really feels clunky. Like if this is a summary at the end, perhaps that makes more sense. I feel like the original text (will paste below) is better and maybe just move it sooner if the goal is to provide an executive summary. 

To set up batching for your custom functions you'll need to write three main sections of code.

A [push operation](https://learn.microsoft.com/en-us/office/dev/add-ins/excel/custom-functions-batching#add-the-_pushoperation-function) to add a new operation to the batch of calls each time Excel calls your custom function.
A [function to make the remote request](https://learn.microsoft.com/en-us/office/dev/add-ins/excel/custom-functions-batching#make-the-remote-request) when the batch is ready.
[Server code to respond to the batch request](https://learn.microsoft.com/en-us/office/dev/add-ins/excel/custom-functions-batching#process-the-batch-call-on-the-remote-service), calculate all of the operation results, and return the values.

Author: Thanks for the great feedback! I rewrote this section and reorganized the first few paragraphs of the article to incorporate your suggestions. Does this seem more accurate?
